### DISCUSSION TITLE
Adding the same image on top of different .gif files

### DISCUSSION THREAD

**Discussion by Ellandar456** [ImageMagick/ImageMagick#4072] [Help] [answered]
Hello,

I'm not sure if this is possible, but I would like to do the following:

- I have a folder filled with .gif files. All sizes are 600x600 px
- I want to add 1 specific image (.jpg, 600x150 px) on the bottom of all the .gif files. Preferrably while making the .gif files 600x750 px (so that nothing is cut from the .gif files).  
- End result should be a .gif file with static .jpg image on the bottom
- I attached an image to make this more clear. 
-
![Combining gif and image](https://user-images.githubusercontent.com/88846134/129230034-9a9cd72f-5847-4a4d-bf3b-68e9960a5c26.jpg)

Would this be possible?

**Comment by snibgo** [comment]
What version of IM, on what platform?

Each output file will be either a GIF, or a JPEG. You have to decide which you want.

The basic command could be:
```
magick in01.gif in.jpg -append +repage out.jpg
```

**Reply by Ellandar456** [reply]
Hello,

Thanks for your reply! My version is Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18 and platform would be Windows 10.

Output should be a GIF 

Thanks!

**Comment by snibgo** [comment]
I should check: I assume your input GIF files each have a single image? If they are animated, the command will be more complex.

I assume you want the output GIFs in another directory, which I will call "outdir".

Assuming they are single images, then:
```
magick in01.gif in.jpg -append +repage outdir\out.gif
```
Use your own filenames, obviously. Does that do what you want?

If it does, you can put it inside a "for" loop like this (untested):
```
for %F in (*.gif) do magick %F in.jpg -append +repage outdir\%F
```
If this is in a BAT script, double the percent signs: %%

**Reply by Ellandar456** [reply]
Thanks for your help, much appreciated!

My GIF files are indeed animated, so it didn't quite work. As a test I tried the command with a non-animated GIF and that worked out perfectly. So i'm guessing it should work with an animated GIF with some more complex command.

If you can spare some time, what would the command be for animated GIF files?

Again, thanks a lot!

**Comment by snibgo** [comment]
```
magick ^
  in.gif ^
  -layers coalesce ^
  -background None -gravity South -splice 0x150+0+0 ^
  NULL: ^
  in.jpg ^
  -compose Over -layers composite ^
  -layers optimize ^
  out.gif
```
Your input GIF is probably optimized, meaning the each frame in the file carries only the pixels that have changed from the previous frame. "-layers coalesce" undoes this, so each frame is a complete image.

We splice 150 transparent rows to the bottom of each frame, making the frame larger. The background colour isn't important.

We add a "NULL:" image to the list, and then the input JPG that is 150 pixels high, so "-compose Over -layers composite" will compose the image from the JPG over every frame from the GIF, at the bottom.

Finally, "-layers optimize" will try to optimize the frames, and we write the output GIF.

**Reply by Ellandar456** [reply]
Works like a charm! As a last question, how should I change the command so that all files are converted in a loop? So without me having to change the file names manually every time (there are thousands of files to change :-s)

Many thanks

**Comment by snibgo** [comment, marked as answer]
Put it inside a "for" loop, as I showed above. For example (Windows BAT syntax):
```
for %%F in (*.gif) do magick
  %%F ^
  -layers coalesce ^
  -background None -gravity South -splice 0x150+0+0 ^
  NULL: ^
  in.jpg ^
  -compose Over -layers composite ^
  -layers optimize ^
  outdir\%%F
```
Note that "outdir" must already exist. IM won't create it.

**Reply by Ellandar456** [reply]
Thank you so much! This will save me hours of adjusting things manually. ImageMagick is exactely the tool I needed for this. Thanks for sharing you knowledge
